buildroot-external/board/generic-aarch64/grub.cfg

```diff
@@ -0,0 +1,61 @@
+set default="0"
+set timeout="5"
+
+set ORDER="A B"
+set A_OK=0
+set B_OK=0
+set A_TRY=0
+set B_TRY=0
+set MACHINE_ID=""
+load_env
+
+# select bootable slot
+for SLOT in $ORDER; do
+    if [ "$SLOT" == "A" ]; then
+        INDEX=1
+        OK=$A_OK
+        TRY=$A_TRY
+        A_TRY=1
+    fi
+    if [ "$SLOT" == "B" ]; then
+        INDEX=2
+        OK=$B_OK
+        TRY=$B_TRY
+        B_TRY=1
+    fi
+    if [ "$OK" -eq 1 -a "$TRY" -eq 0 ]; then
+        default=$INDEX
+        break
+    fi
+done
+
+# reset booted flags
+if [ "$default" -eq 0 ]; then
+    if [ "$A_OK" -eq 1 -a "$A_TRY" -eq 1 ]; then
+        A_TRY=0
+    fi
+    if [ "$B_OK" -eq 1 -a "$B_TRY" -eq 1 ]; then
+        B_TRY=0
+    fi
+fi
+
+save_env A_TRY A_OK B_TRY B_OK ORDER MACHINE_ID
+
+CMDLINE="rootwait zram.enabled=1 zram.num_devices=3 apparmor=1 security=apparmor rootwait systemd.machine_id=$MACHINE_ID cgroup_enable=memory fsck.repair=yes"
+
+# root is a full HDD/partition definition in GRUB format like hd0,gpt1
+# We extract the part before the comma to then append our own partition index
+# at the end. This is hacky but the best way I found
+regexp --set 1:boothd (.+),.+ ${root}
+
+menuentry "Slot DEFAULT (for RAUC to fall back onto)" {
+    linux (${boothd},gpt2)/Image root=PARTUUID=8d3d53e3-6d49-4c38-8349-aff6859e82fd $CMDLINE rauc.slot=A
+}
```

Comment:
I guess technically that is not even necessary is it? It would fallback to the first entry which is A as well..?